lib/Mojolicious/Guides/Tutorial.pod

```diff
@@ -0,0 +1,938 @@
+
+  use Mojolicious::Lite -signatures;
+
+  # Load message into memory
+  my $hello = app->home->child('cache', 'hello.txt')->slurp;
+
+  # Display message
+  get '/' => sub ($c) {
+    $c->render(text => $hello);
+  };
+
+You can also introspect your application from the command line with L<Mojolicious::Command::eval>.
+
+  $ ./myapp.pl eval -v 'app->home'
+
+=head2 Conditions
+
+Conditions such as C<agent> and C<host> from L<Mojolicious::Plugin::HeaderCondition> allow even more powerful route
+constructs.
+
+  use Mojolicious::Lite -signatures;
+
+  # Firefox
+  get '/foo' => (agent => qr/Firefox/) => sub ($c) {
+    $c->render(text => 'Congratulations, you are using a cool browser.');
+  };
+
+  # Internet Explorer
+  get '/foo' => (agent => qr/Internet Explorer/) => sub ($c) {
+    $c->render(text => 'Dude, you really need to upgrade to Firefox.');
+  };
+
+  # http://mojolicious.org/bar
+  get '/bar' => (host => 'mojolicious.org') => sub ($c) {
+    $c->render(text => 'Hello Mojolicious.');
+  };
+
+  app->start;
+
+=head2 Sessions
+
+Cookie-based sessions just work out of the box, as soon as you start using them through the helper
+L<Mojolicious::Plugin::DefaultHelpers/"session">. Just be aware that all session data gets serialized with
+L<Mojo::JSON> and stored client-side, with a cryptographic signature to prevent tampering.
+
+  use Mojolicious::Lite -signatures;
+
+  # Access session data in action and template
+  get '/counter' => sub ($c) {
+    $c->session->{counter}++;
+  };
+
+  app->start;
+  __DATA__
+
+  @@ counter.html.ep
+  Counter: <%= session 'counter' %>
+
+Note that you should use custom L<Mojolicious/"secrets"> to make signed cookies really tamper resistant.
+
+  app->secrets(['My secret passphrase here']);
+
+=head2 File uploads
+
+All files uploaded via C<multipart/form-data> request are automatically available as L<Mojo::Upload> objects from
+L<Mojolicious::Controller/"param">. And you don't have to worry about memory usage, because all files above 250KiB will
+be automatically streamed into a temporary file. To build HTML forms more efficiently, you can also use tag helpers
+like L<Mojolicious::Plugin::TagHelpers/"form_for">.
+
+  use Mojolicious::Lite -signatures;
+
+  # Upload form in DATA section
+  get '/' => 'form';
+
+  # Multipart upload handler
+  post '/upload' => sub ($c) {
+
+    # Check file size
+    return $c->render(text => 'File is too big.', status => 200) if $c->req->is_limit_exceeded;
+
+    # Process uploaded file
+    return $c->redirect_to('form') unless my $example = $c->param('example');
+    my $size = $example->size;
+    my $name = $example->filename;
+    $c->render(text => "Thanks for uploading $size byte file $name.");
+  };
+
+  app->start;
+  __DATA__
+
+  @@ form.html.ep
+  <!DOCTYPE html>
+  <html>
+    <head><title>Upload</title></head>
+    <body>
+      %= form_for upload => (enctype => 'multipart/form-data') => begin
```

Comment:
Is it documented somewhere that (method => 'POST') is only needed on `any` routes?  i.e. the method will be automatically selected based on a registered route definition.  Document it, or link to hit from here.